app/i18n/translator.html

```diff
@@ -128,111 +165,280 @@
 
-        const prevStr = state.prevOriginal[state.selectedKey] ? state.prevOriginal[state.selectedKey] : "";
+        if (state.showDiff && origStr !== "" && prevStr !== "") {
+          var d = dmp.diff_main(prevStr, origStr);
+          dmp.diff_cleanupEfficiency(d);
+          elID("original_str").innerHTML = dmp.diff_prettyHtml(d);
+        } else {
+          elID("original_str").innerHTML = origStr;
+        }
         elID("prev_original_str").innerText = prevStr;
-        elID("prev_original_str").hidden = (prevStr == "") || (origStr == prevStr);
+        if (prevStr === "" || origStr === prevStr) {
+          elID("toggle_diff_bt").disabled = true;
+          elID("prev_original_str").classList.add("inactive");
+        } else {
+          elID("prev_original_str").classList.remove("inactive");
+          elID("toggle_diff_bt").disabled = false;
+        }
+        elID("key_str").innerText = state.selectedKey;
+
+        elID("prev_bt").innerText = isDeleted ? "previous" : "save & previous";
+        elID("prev_bt").disabled = state.previousKeys.length <= 1;
+        elID("copy_bt").disabled = isDeleted || origStr === "";
+        elID("next_bt").disabled = isDeleted || state.selectedKey === "";
+        elID("translated_str").disabled = isDeleted || state.selectedKey === "";
+        elID("toggle_diff_bt").innerText = state.showDiff
+          ? "Hide diff"
+          : "Show diff";
+      }
+
+      function getIndexOfSelectedKey(keys) {
+        return Math.max(0, keys.indexOf(state.selectedKey));
       }
 
       function nextKey() {
         let gotPrev = false;
         let keys = Object.keys(state.original);
-        const filterVal = document.querySelector('input[name="filter"]:checked').value;
-        for (let i = 0; i < keys.length; i++) {
+        let filterVal = getFilterVal();
+        const { filterVal: previousFilterVal } =
+          state.previousKeys.length > 0 &&
+          state.previousKeys[state.previousKeys.length - 1];
+        if (previousFilterVal !== undefined && filterVal != previousFilterVal) {
+          filterVal = previousFilterVal;
+          setFilterVal(filterVal);
+        }
+
+        for (let i = getIndexOfSelectedKey(keys); i < keys.length; i++) {
           const k = keys[i];
           // When on Untranslated view, skip to next key considered
           // "untranslated", i.e. undefined (new key) or "" (existing key with
           // empty translation).
           if (filterVal === "untranslated" && state.translated[k]) {
             continue;
           }
+          // When on Changed view, skip to next key considered "changed".
+          if (
+            filterVal === "changed" &&
+            state.changedEntries.indexOf(k) === -1
+          ) {
+            continue;
+          }
 
           // When on Changed view, skip to next key considered "changed".
           if (filterVal === "changed" && state.changedEntries.indexOf(k) == -1) {
             continue;
           }
```

Comment:
Looks like a duplicate of the previous code block. Better remove this one since it has one remaining `==`.